Add minimum python version to recipe.

tompollard opened the pull request:
https://github.com/conda-forge/tableone-feedstock/pull/37 is failing on conda-forge-linter with the following message: 

```
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (recipe) and found some lint.

Here's what I've got...

For recipe:

noarch: python recipes are required to have a lower bound on the python version. Typically this means putting python >=3.6 in both host and run but you should check upstream for the package's Python compatibility.
```

This pull request is an attempt to fix the issue, by setting a lower bound on the Python version to `python >=3.9`. Also increase the build number as suggested in https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests.

tompollard committed:
add minimum python version to recipe.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

tompollard committed:
manually set ubuntu image

according to the file, updates should be made to conda-forge.yml and/or the recipe/meta.yaml, but I don't see where this could be set in those files.
